docs/basics/part-6-using-docker-with-earthly.md

```diff
@@ -0,0 +1,565 @@
+  java/part6_default:
+
+
+```
+The code of the app might look like this
+
+`./src/main/java/hello/HelloWorld.java`
+
+```java
+
+package postgresclient;
+
+import org.joda.time.LocalTime;
+import java.sql.Connection;
+import java.sql.DriverManager;
+
+
+public class PostgreSQLJDBC {
+   public static void main(String args[]) {
+      Connection c = null;
+      try {
+         Class.forName("org.postgresql.Driver");
+         c = DriverManager
+            .getConnection("jdbc:postgresql://postgres:5432/test_db",
+            "earthly", "password");
+      } catch (Exception e) {
+         e.printStackTrace();
+         System.err.println(e.getClass().getName()+": "+e.getMessage());
+         System.exit(0);
+      }
+      System.out.println("Opened database successfully");
+   }
+}
+```
+
+`./build.gradle`
+
+```groovy
+apply plugin: 'java'
+apply plugin: 'application'
+
+mainClassName = 'postgresclient.PostgreSQLJDBC'
+
+repositories {
+    mavenCentral()
+}
+
+jar {
+    baseName = 'hello-world'
+    version = '0.0.1'
+}
+
+sourceCompatibility = 1.8
+targetCompatibility = 1.8
+
+dependencies {
+    compile "joda-time:joda-time:2.2"
+    compile(group: 'org.postgresql', name: 'postgresql', version: '42.3.3')
+    testCompile "junit:junit:4.12"
+}
+
+```
+</details>
+
+<details open>
+<summary>Python</summary>
+
+To copy the files for [this example ( Part 7 )](https://github.com/earthly/earthly/tree/main/examples/tutorial/python/part7) run
+
+```bash
+earthly --artifact github.com/earthly/earthly/examples/tutorial/python:main+part7/part7 ./part7
+```
+`./tests/test_db_connection.py`
+
+```python
+import unittest
+import psycopg2
+
+class MyIntegrationTests(unittest.TestCase):
+
+    def test_db_connection_active(self):
+        connection = psycopg2.connect(
+            host="postgres",
+            database="test_db",
+            user="earthly",
+            password="password")
+        
+        self.assertEqual(connection.closed, 0)
+
+if __name__ == '__main__':
+    unittest.main()
+```
+
+```yml
+version: "3.9"
+   
+services:
+  db:
+    image: postgres
+    container_name: db
+    hostname: postgres
+    environment:
+      - POSTGRES_DB=test_db
+      - POSTGRES_USER=earthly
+      - POSTGRES_PASSWORD=password
+    ports:
+      - 5432:5432
+    networks:
+      - python/part6_default
+
+networks:
+  python/part6_default:
+```
+
+`./Earthfile`
+
+```Dockerfile
+VERSION 0.6
+FROM python:3
+WORKDIR /code
+
+build:
+  COPY ./requirements.txt .
+  RUN pip install -r requirements.txt
+  COPY . .
+
+run-tests:
+    FROM earthly/dind:alpine
+    COPY ./docker-compose.yml .
+    COPY ./tests ./tests
+    RUN apk update
+    RUN apk add postgresql-client
+    WITH DOCKER --compose docker-compose.yml --load app:latest=+docker
+        RUN while ! pg_isready --host=localhost --port=5432; do sleep 1; done ;\
+          docker run --network=default_python/part6_default app python3 ./tests/test_db_connection.py
+    END
+```
+</details>
```

Comment:
The examples are great! I assume you tested them locally and they all work, right?

Reply:
Yes, they all run locally. Just tested them again to be sure :)